src/hotspot/share/utilities/count_trailing_zeros.hpp

```diff
@@ -59,13 +65,23 @@ inline unsigned count_trailing_zeros(uintx x) {
 #pragma intrinsic(_BitScanForward)
 #endif
 
-inline unsigned count_trailing_zeros(uintx x) {
-  assert(x != 0, "precondition");
+inline unsigned count_trailing_zeros_32(uint32_t x) {
+  unsigned long index;
+  _BitScanForward(&index, x);
+  return index;
+}
+
+inline unsigned count_trailing_zeros_64(uint64_t x) {
   unsigned long index;
 #ifdef _LP64
   _BitScanForward64(&index, x);
 #else
-  _BitScanForward(&index, x);
+  if (_BitScanForward(&index, (uint32_t)x) == 0) {
+    // no bit found? If so, try the upper dword. Otherwise index already contains the result
+    _BitScanForward(&index, x >> 32);
+    assert(index > 0, "invariant since x != 0");
```

Comment:
I'm not sure this assert really does anything useful.  If both of the BSFs failed because the value is zero (which shouldn't happen because we checked for that), then the value of index is undefined.